src/server/pfs/server/master.go

```diff
@@ -70,65 +71,78 @@ func (d *driver) finishCommits(ctx context.Context) error {
 			return nil
 		}
 		commit := commitInfo.Commit
-		return backoff.RetryUntilCancel(ctx, func() error {
-			id, err := d.getFileSet(ctx, commit)
-			if err != nil {
-				if pfsserver.IsCommitNotFoundErr(err) {
-					return nil
-				}
-				return err
-			}
-			// Compact the commit.
-			start := time.Now()
-			totalId, err := d.compactor.Compact(ctx, []fileset.ID{*id}, defaultTTL)
-			if err != nil {
-				return err
-			}
-			if err := d.commitStore.SetTotalFileSet(ctx, commit, *totalId); err != nil {
-				return err
-			}
-			compactingDuration := time.Since(start)
-			// Validate the commit.
-			start = time.Now()
-			size, validationError, err := d.validate(ctx, totalId)
-			if err != nil {
-				return err
-			}
-			validatingDuration := time.Since(start)
-			// Finish the commit.
-			return d.txnEnv.WithWriteContext(ctx, func(txnCtx *txncontext.TransactionContext) error {
-				commitInfo := &pfs.CommitInfo{}
-				if err := d.commits.ReadWrite(txnCtx.SqlTx).Update(pfsdb.CommitKey(commit), commitInfo, func() error {
-					commitInfo.Finished = txnCtx.Timestamp
-					commitInfo.SizeBytesUpperBound = size
-					if commitInfo.Details == nil {
-						commitInfo.Details = &pfs.CommitInfo_Details{}
+		return miscutil.LogStep(fmt.Sprintf("finishing commit %v", commit.ID), func() error {
```

Comment:
The addition of the callback makes this diff pretty big. I'm under the impression nothing changed here though other than the addition of logging?

Reply:
Yeah.